stl/inc/xutility

```diff
@@ -2843,18 +2843,18 @@ namespace ranges {
         using _Not_quite_object::_Not_quite_object;
 
         // clang-format off
-        template <input_or_output_iterator _It, sentinel_for<_It> _Se>
+        template <class _It, sentinel_for<_It> _Se>
             requires (!sized_sentinel_for<_Se, _It>)
         _NODISCARD constexpr iter_difference_t<_It> operator()(_It _First, _Se _Last) const {
             // clang-format on
             _Adl_verify_range(_First, _Last);
             return _Distance_unchecked(_Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)));
         }
 
-        template <input_or_output_iterator _It, sized_sentinel_for<_It> _Se>
-        _NODISCARD constexpr iter_difference_t<_It> operator()(const _It& _First, const _Se& _Last) const
-            noexcept(noexcept(_Last - _First)) /* strengthened */ {
-            return _Last - _First;
+        template <class _It, sized_sentinel_for<decay_t<const _It>> _Se>
+        _NODISCARD constexpr iter_difference_t<decay_t<const _It>> operator()(const _It& _First, _Se _Last) const

```

Comment:
As I mentioned in the LWG thread, I think this should use `_It&&` and then `decay_t<_It>`, to avoid turning a non-const array into a pointer-to-const.

Reply:
Thanks!